lib/wallet/http.js

```diff
@@ -833,6 +784,473 @@ class HTTP extends Server {
+
+    // Wallet Bids by Name
+    this.get('/wallet/:id/bid/:name', async (req, res) => {
+      const valid = Validator.fromRequest(req);
+      const name = valid.str('name');
+      let own = valid.bool('own', false);
+
+      if (name)
+        assert(rules.verifyName(name), 'Must pass valid name.');
+
+      if (!name)
+        own = true;
+
+      const bids = await req.wallet.getBidsByName(name);
+      const items = [];
+
+      for (const bid of bids) {
+        if (!own || bid.own)
+          items.push(bid.toJSON());
+      }
+
+      res.json(200, items);
+    });
+
+    // All Wallet Reveals
+    this.get('/wallet/:id/reveal', async (req, res) => {
+      const valid = Validator.fromRequest(req);
+      const own = valid.bool('own', false);
+
+      const reveals = await req.wallet.getRevealsByName();
+      const items = [];
+
+      for (const brv of reveals) {
+        if (!own || brv.own)
+          items.push(brv.toJSON());
+      }
+
+      res.json(200, items);
+    });
+
+    // Wallet Reveals by Name
+    this.get('/wallet/:id/reveal/:name', async (req, res) => {
+      const valid = Validator.fromRequest(req);
+      const name = valid.str('name');
+      let own = valid.bool('own', false);
+
+      if (name)
+        assert(rules.verifyName(name), 'Must pass valid name.');
+
+      if (!name)
+        own = true;
+
+      const reveals = await req.wallet.getRevealsByName(name);
+      const items = [];
+
+      for (const brv of reveals) {
+        if (!own || brv.own)
+          items.push(brv.toJSON());
+      }
+
+      res.json(200, items);
+    });
+
+    // Name Resource
+    this.get('/wallet/:id/resource/:name', async (req, res) => {
+      const valid = Validator.fromRequest(req);
+      const name = valid.str('name');
+
+      assert(name, 'Must pass name.');
+      assert(rules.verifyName(name), 'Must pass valid name.');
+
+      const ns = await req.wallet.getNameStateByName(name);
+
+      if (!ns || ns.data.length === 0)
+        return res.json(404);
+
+      try {
+        const resource = Resource.decode(ns.data);
+        return res.json(200, resource.toJSON());
+      } catch (e) {
+        return res.json(400);
+      }
+    });
+
+    // Regenerate Nonce
+    this.get('/wallet/:id/nonce/:name', async (req, res) => {
+      const valid = Validator.fromRequest(req);
+      const name = valid.str('name');
+      const addr = valid.str('address');
+      const bid = valid.ufixed('bid');
+
+      assert(name, 'Name is required.');
+      assert(rules.verifyName(name), 'Valid name is required.');
+      assert(addr, 'Address is required.');
+      assert(bid, 'Bid is required.');
+
+      let address;
+      try {
+        address = Address.fromString(addr, this.network);
+      } catch (e) {
+        return req.json(400);
+      }
+
+      const nameHash = rules.hashName(name);
+      const nonce = await req.wallet.generateNonce(nameHash, address, bid);
+      const blind = rules.blind(bid, nonce);
+
```

Comment:
What is a useful JSON API to return in this case? I think that this is returning all of the useful information a client would need to see the blind and the nonce and then be able to recalculate them client side if they are interested in doing so.

Reply:
Because we are returning the `name` and `bid,` even though the request provides them, why are we not also returning the address? I think we should either return them all or assume the client has the information already.

Reply:
Added `address` as a property on the returned object